creativecommons/search
# Implement Smooth Dropdown Animation for Explore Button

## Fixes
- Fixes #239  by @mo-gbolahan 

## Description
This PR resolves the issue where the dropdown for the Explore button appeared abruptly, creating a poor user experience. The transition property doesn’t work with the display property when toggling between none and block, which is why I replaced display with a combination of translate, height, opacity, and padding to achieve a smooth dropdown effect.

## Technical details

1. Replaced the display property (which cannot transition smoothly) with:
- translate for sliding position animation.
- height and padding for smooth expansion and collapse of the dropdown.
- opacity for a fade-in/fade-out effect.


## Tests
1. Click the Explore button in the top right corner of the page.
2. Observe the dropdown smoothly sliding into view with a fade effect.
3. Click the button again to hide the dropdown, ensuring the transition is smooth when collapsing as well.



## Checklist



  index.md`).
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin




For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
